contracts/consensus/Consensus.sol

```diff
@@ -608,46 +636,56 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
         reputation.deregister(msg.sender);
     }
 
-    /**
-     * @notice Creates a new meta chain given an anchor.
+    /** @notice Creates a new meta chain.
      *         This can be called only by axiom.
      *
      * @dev Function requires:
      *          - msg.sender should be axiom contract.
      *          - core is not assigned to metachain.
-     *
-     * @param _anchor anchor of the new meta-chain.
-     * @param _epochLength Epoch length for new meta-chain.
-     * @param _rootBlockHeight root block height.
      */
-    function newMetaChain(
-        address _anchor,
-        uint256 _epochLength,
-        uint256 _rootBlockHeight
-    )
+    function newMetaChain()
         external
         onlyAxiom
+        returns(bytes32 metachainId_)
     {
-        bytes32 metachainId = hashMetachainId(_anchor);
 
-        require(
-            assignments[metachainId] == address(0),
-            "A core is already assigned to this metachain."
+        bytes memory anchorSetupCallData = anchorSetupData(
+            ANCHOR_MAX_STATE_ROOTS,
+            address(this)
         );
 
-        address core = newCore(
-            metachainId,
-            _epochLength,
+        address anchor = axiom.deployAnchor(anchorSetupCallData);
+        metachainId_ = hashMetachainId(anchor);
+
+        bytes memory coreSetupCallData = coreSetupData(
+            metachainId_,
+            EPOCH_LENGTH,
             uint256(0), // metablock height
             bytes32(0), // parent hash
             gasTargetDelta, // gas target
             uint256(0), // dynasty
             uint256(0), // accumulated gas
-            _rootBlockHeight
+            0 // todo should be fixed with remove source block height in core-setup.
```

Comment:
Source block height will not be removed.
It was a source block hash, and its already removed.